### PR TITLE
Issue #922: spec: run-spec.sh の effort 再校正 (Sonnet 5 xhigh 候補、Opus fallback 比較、C3)

### LINKED ISSUE #922: spec: run-spec.sh の effort 再校正 (Sonnet 5 xhigh 候補、Opus fallback 比較、C3)
## Background

`docs/reports/claude-sonnet-5-impact-strategy.md` §8 の候補 Issue **C3**。C1 (#914、default parent を Sonnet 5 に切替) が着地し、effort 再校正が実行可能になった。

現行の matrix (`docs/tech.md`):
- `run-spec.sh` (spec): Sonnet **max** (L サイズは `--opus` で Opus xhigh/max、`--fable` で Fable 5)

レポート §4.2 の再校正候補:
- **run-spec.sh**: 設計品質にクリティカルなフェーズ (spec エラーは全下流フェーズに伝播)。§3.3 の「Sonnet 5 `xhigh` が Opus 4.8 に迫る」というフレーミングは、L サイズ spec で Opus に落とさずとも `xhigh` で十分かもしれないことを示唆する。ただし品質感度が高く、default の切替ではなく慎重な評価を要する。

## Purpose

`run-spec.sh` の effort について、Sonnet `max` → `xhigh` へ下げられるか、および L サイズでの Opus fallback (`--opus`) との品質比較を評価し、判定と根拠を `docs/tech.md` matrix に記録する。品質クリティカルなフェーズのため、A/B の品質評価を重視する。

### Pre-merge (auto-verified)

- [x]  spec effort の再校正判定 (Opus fallback 比較を含む) と根拠が docs/tech.md に記録されている
- [x]  変更時は run-spec.sh と matrix 表が SSoT として一致している

### Post-merge

なし

## Notes

- **CI-sensitive**: run-spec.sh の effort 値と matrix 表 (SSoT) に触れるため、レポート §8 では Size S 想定だが `[[feedback_ci_sensitive_size_m]]` に従い PR route (Size M 相当) で実行するのが安全。
- **品質クリティカル**: spec エラーは全下流に伝播するため、C2/C4 より慎重な評価が必要。default の切替ではなく十分な A/B 品質評価を伴うこと。
- C2 (run-code/run-review)・C4 (run-issue) とは独立。

## Reference

- `docs/reports/claude-sonnet-5-impact-strategy.md` §3.3 / §4.2 (run-spec.sh 候補) / §8 (C3)
- #914 (C1、完了済み前提)
- `docs/tech.md` §Phase-specific model and effort matrix (`ssot_for: model-effort-matrix`、更新対象)

## Auto-Resolved Ambiguity Points

非対話モード (`--non-interactive`) 実行のため、以下の曖昧ポイントをモデル判断で自動解決した (詳細は Issue Retrospective コメントを参照):

1. **A/B 品質評価の方法論**: 本リポジトリには spec 生成品質を比較する専用の実測計装が存在しない。#877 (`/verify` Sonnet 5 再測定) の先例に倣い、既存資料 (レポート §3.3 の効果曲線に関する記述、`docs/tech.md` 内の既存 rationale、および可能な範囲での実サンプル比較) に基づく推論的評価を採用し、代理的根拠であることの限界を判定記録に明記する方針とする。
2. **Opus fallback との比較範囲**: AC1 の文言「Sonnet max → xhigh 候補の評価結果と、Opus fallback との比較を踏まえた判定」から、現行の Sonnet max を基準線とした 3-way 比較 (Sonnet max 現行 / Sonnet xhigh 候補 / Opus xhigh・max) と解釈する。AC 文言自体で既に一意に読み取れるため、Issue 本文の変更は不要と判断した。
3. **「据え置き」判定時の記録場所**: AC2 は「据え置いた場合は表に変更がない」とするが、AC1 は判定方向によらず根拠の記録を要求する。`docs/tech.md` 既存の "Sonnet 5" ノートや "Opus 4.8 effort calibration" ノートと同様、表セル自体は変更せず、表下の自由記述ノートとして根拠を追記する運用を踏襲する。

### PR DESCRIPTION
**Pull request by saito**
## Summary
- Evaluated whether `run-spec.sh`'s Sonnet-path default effort can drop from `max` to `xhigh` under Sonnet 5's widened effort curve (impact strategy report §3.3/§4.2, C3), including a comparison against the existing Opus fallback (`--opus`, L-size only, `xhigh` default since #217)
- **Verdict: maintain `max`** — no Anthropic guidance documents a `max`-effort diminishing-returns risk for Sonnet 5 specifically (the only such guidance recorded in `docs/tech.md`, the "Opus 4.8 effort calibration" note, is scoped to Opus 4.8); `/spec` is structurally analogous to `run-code.sh` (single-agent, no sub-agent fan-out) but carries strictly higher downstream blast radius since spec errors propagate through code, review, *and* merge
- New analysis report: `docs/reports/sonnet-5-effort-recalibration-spec.md` (structural comparison with #921's precedent, precedent re-derivation of #217/#229, production-sample check of Code Retrospectives since #914)
- Recorded the verdict as a prose note in `docs/tech.md` § Phase-specific model and effort matrix, mirrored to `docs/ja/tech.md`
- No changes to `scripts/run-spec.sh` or `tests/run-spec.bats` (consistent with the "maintain" verdict — matrix table cells and script effort value are unchanged, per the Issue's own Auto-Resolved Ambiguity Point #3)

## Verification (pre-merge)
- AC1 (rubric): spec effort recalibration verdict (including Opus fallback comparison) and rationale recorded in `docs/tech.md` — satisfied by the new prose note
- AC2 (rubric): SSoT consistency — verdict is "maintain," so `run-spec.sh` and the matrix table are unchanged and remain consistent; `tests/run-spec.bats`'s 32 tests pass locally, confirming no regression

## Verification (post-merge)
- None (Issue AC has no post-merge items)

closes #922

### CODE REVIEW

File: docs/reports/sonnet-5-effort-recalibration-spec.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Sonnet 5 Effort Recalibration: `run-spec.sh`
+
+**Report date**: 2026-07-05
+**Issue**: #922 (C3, `docs/reports/claude-sonnet-5-impact-strategy.md` §8)
+**Scope**: Re-evaluate whether `run-spec.sh`'s Sonnet-path default effort can drop from `max` to `xhigh` under Sonnet 5's widened effort curve, including a comparison against the existing Opus fallback (`--opus`, L-size only).
+**Depends on**: `docs/reports/claude-sonnet-5-impact-strategy.md` §3.3/§4.2/§4.6 (candidate framing); Issue #217 (2026-xx-xx, Opus path `max`→`xhigh` swap — Sonnet path explicitly out of scope); Issue #229 (`docs/reports/sonnet-effort-recalibration.md`, Sonnet 4.6-era evaluation of `run-code.sh`/`run-review.sh`/`run-merge.sh`/`run-verify.sh`/`run-issue.sh` — `run-spec.sh` explicitly excluded); `docs/reports/sonnet-5-effort-recalibration-code-review.md` (#921, C2, sibling report and methodology precedent).
```
Comment (saito) [inline docs/reports/sonnet-5-effort-recalibration-spec.md]: [Documentation Consistency] SHOULD: unfilled placeholder date `2026-xx-xx` for Issue #217 in the "Depends on" line. Issue #217's actual `closedAt` is `2026-04-17` (matching the dating convention used for #229 elsewhere in this same document family, e.g. `docs/tech.md` dates #229 as `2026-04-18`). Recommended fix: replace `2026-xx-xx` with `2026-04-17`.

File: docs/reports/sonnet-5-effort-recalibration-spec.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+Surveyed `docs/spec/*.md` Code Retrospective "Design Gaps/Ambiguities" sections for Issues numbered ≥ 915 (coded after #914 landed the Sonnet 5 default-parent swap), the population for which any `run-spec.sh` Sonnet-path quality signal would necessarily come from Sonnet 5 itself:
+
+| Issue | Code Retrospective "Design Gaps/Ambiguities" |
+|---|---|
+| #915 | N/A |
+| #916 | N/A |
+| #917 | Two entries — both about an emergent interaction between a recovery-guidance example and `validate-recovery-plan.sh`'s forbidden-push-pattern regex / the auto-mode security classifier, discovered only by running the actual bats test and hitting the classifier live. Not a spec-design-reasoning gap; the constraint (regex behavior + live classifier response) was not knowable from the Issue/Spec context alone. |
+| #918 | N/A |
+| #921 | One entry — a process/procedural note about resuming a previously-interrupted `/code` run (label-state investigation), not a spec-content quality issue. |
+| #927 | (Code phase not yet run at report time — no data) |
```
Comment (saito) [inline docs/reports/sonnet-5-effort-recalibration-spec.md]: [Edge Case/Robustness] SHOULD: the production-sample table claims "#927 | (Code phase not yet run at report time — no data)", but `docs/spec/issue-927-review-phase-silent-noop.md`'s Code Retrospective (`### Design Gaps/Ambiguities`) was already committed (commit `96ae9bc2`, 2026-07-05 19:03) well before this report's own commit — the survey misrepresents its own evidence base for this row. Doesn't change the verdict (an actual "N/A" entry would have reinforced, not undermined, the "no design-reasoning gaps" conclusion), but undermines the report's evidentiary integrity as an audit artifact. Recommended fix: update the row to reflect the actual #927 Code Retrospective content ("N/A — no design-reasoning gap") instead of "no data".

**Review by saito — COMMENTED** [review]
## Acceptance Criteria Verification Results

| Condition | Result | Notes |
|-----------|--------|-------|
| spec effort の再校正判定 (Opus fallback 比較を含む) と根拠が docs/tech.md に記録されている | PASS | `docs/tech.md` § Phase-specific model and effort matrix に「Sonnet 5 effort recalibration — spec (#922, C3)」ノートとして判定 (維持)・Opus fallback 比較・根拠が記録されている |
| 変更時は run-spec.sh と matrix 表が SSoT として一致している (据え置き時は表に変更なし) | PASS | 判定は「据え置き」。`scripts/run-spec.sh`/`tests/run-spec.bats` は本 PR で無変更であることを確認 (`EFFORT="max"` のまま) |

## CI Status

| Job name | Status | Notes |
|---------|--------|-------|
| DCO | SUCCESS | — |
| Run bats tests | SUCCESS | — |
| Validate skill syntax | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## Code Review

### Lightweight Integrated Review (review-light, all 4 aspects)

Spec deviation / Security / Documentation consistency / Edge cases の4観点を1エージェントで実施 (Size M, `--light`)。

- Spec deviation: 問題なし。据え置き判定に伴う実装 (run-spec.sh/tests/run-spec.bats 無変更) が Spec の Implementation Steps と一致
- Security: 問題なし。ドキュメントファイルのみの変更
- Edge cases / Documentation consistency: 下記2件の SHOULD 指摘あり (line comments 参照)

### Issue Count Summary

- Posted as line comments: 2 (MUST: 0, SHOULD: 2, CONSIDER: 0)
- General Comments (in body): 0

---
Next: run `/merge 936`.

**Comment by saito** [comment]
## Review Response Summary

### Claude Review Response

| Aspect | Filename | Line | Issue | Response | Fix/Skip reason |
|--------|---------|------|-------|----------|----------------|
| Documentation consistency | docs/reports/sonnet-5-effort-recalibration-spec.md | 6 | Unfilled placeholder date `2026-xx-xx` for Issue #217 | Resolved | Replaced with actual `closedAt` date `2026-04-17` |
| Edge cases/Robustness | docs/reports/sonnet-5-effort-recalibration-spec.md | 59 | Production-sample table claimed "#927: no data" despite #927's Code Retrospective already being committed at authoring time | Resolved | Updated row to "N/A — no design-reasoning gap recorded"; also updated the "four populated samples" summary sentence to "five" to stay consistent |

### Lightweight Re-check Results

- `python3 scripts/validate-skill-syntax.py skills/`: 0 error(s), 1 pre-existing warning (`skills/auto/SKILL.md` — unrelated to this PR's changes)
- Fix diff check: PASS (both edits are isolated text corrections; no new issues introduced)
- New issues: none

---
Next: run `/merge 936`.
